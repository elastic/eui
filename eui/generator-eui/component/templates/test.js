import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  test('is rendered', () => {
    const component = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
