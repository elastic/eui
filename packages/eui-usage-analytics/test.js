/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const Codeowners = require('codeowners');
const temp = new Codeowners("../../../kibana");

const { scan } = require('./scan');

const runScan = async () => {
  const scanResult = await scan();
  console.log(scanResult);
};

runScan();
