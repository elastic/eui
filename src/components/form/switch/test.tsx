import React from 'react';

import { EuiSwitch } from './switch';

export const Test = ({ checked }: { checked: boolean }) => (
  <EuiSwitch
    checked={checked}
    label="Test"
    onChange={e => console.log(e.target.checked)}
  />
);
