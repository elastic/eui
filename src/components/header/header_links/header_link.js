import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { EuiButtonEmpty } from '../../button';

export const EuiHeaderLink = ({
  href,
  onClick,
  iconType,
  isActive,
  children,
  className,
  ...rest,
}) => {
  const classes = classNames('euiHeaderLink', className);

  return (
    <EuiButtonEmpty
      className={classes}
      href={href}
      onClick={onClick}
      iconType={iconType}
      color={isActive ? 'primary' : 'text'}
      {...rest}
    >
      {children}
    </EuiButtonEmpty>
  );
};

EuiHeaderLink.propTypes = {
  href: PropTypes.string,
  children: PropTypes.node,
  isActive: PropTypes.bool,
};
