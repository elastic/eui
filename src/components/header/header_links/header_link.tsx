/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

import { EuiButtonEmpty, EuiButtonEmptyProps } from '../../button';

export type EuiHeaderLinkProps = EuiButtonEmptyProps & {
  /**
   * Simple prop to update color based on active state.
   * Can be overridden with `color`
   */
  isActive?: boolean;
};

export const EuiHeaderLink: FunctionComponent<EuiHeaderLinkProps> = ({
  isActive,
  className,
  ...rest
}) => {
  const classes = classNames(
    'euiHeaderLink',
    {
      'euiHeaderLink-isActive': isActive,
    },
    className
  );

  const props = {
    color: isActive ? 'primary' : 'text',
    ...rest,
    className: classes,
  };

  return <EuiButtonEmpty {...(props as EuiButtonEmptyProps)} />;
};
