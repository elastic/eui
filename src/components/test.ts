export function add(x: number, y: number) {
  return x + y;
}

add(5, 4);
