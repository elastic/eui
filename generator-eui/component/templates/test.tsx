/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  it('renders', () => {
    const { container } = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
