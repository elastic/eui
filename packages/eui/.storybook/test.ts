/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { queries, within, waitFor, fireEvent, expect } from '@storybook/test';
import * as dataTestSubjQueries from '../src/test/rtl/data_test_subj_queries';

/**
 * Custom Storybook within util with EUI query helpers
 * + additional chained async/waitFor component utils
 *
 * @see https://storybook.js.org/docs/writing-stories/play-function#writing-stories-with-the-play-function
 * @see https://testing-library.com/docs/dom-testing-library/api-within/
 */
const customWithin = (canvasElement: HTMLElement) => {
  const canvas = within<typeof queries & typeof dataTestSubjQueries>(
    canvasElement,
    { ...queries, ...dataTestSubjQueries }
  );

  return {
    ...canvas,

    /**
     * 1. Loki doesn't like userEvent, only fireEvent
     * 2. Storybook fires fireEvents too early (esp. on page load), so we add a waitFor
     */
    waitForAndClick: async (testSubject: string) => {
      await waitFor(() =>
        expect(canvas.getByTestSubject(testSubject)).toBeInTheDocument()
      );
      await fireEvent.click(canvas.getByTestSubject(testSubject));
    },

    waitForEuiPopoverVisible: async () =>
      await waitFor(() =>
        expect(canvasElement.querySelector('[data-popover-open]')).toBeVisible()
      ),
    waitForEuiPopoverHidden: async () =>
      await waitFor(() =>
        expect(
          canvasElement.querySelector('[data-popover-panel]')
        ).not.toBeInTheDocument()
      ),
  };
};

export { customWithin as within };
