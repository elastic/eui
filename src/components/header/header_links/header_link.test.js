import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../../test';

import { EuiHeaderLink } from './header_link';

describe('EuiHeaderLink', () => {
  test('is rendered', () => {
    const component = render(
      <EuiHeaderLink {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
