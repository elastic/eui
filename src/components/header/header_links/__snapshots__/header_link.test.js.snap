// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiHeaderLink is rendered 1`] = `
<li
  aria-label="aria-label"
  class="euiHeaderLink testClass1 testClass2"
  data-test-subj="test subject string"
>
  <a
    class="euiHeaderLink__link"
  >
    <span
      class="euiHeaderLink__text"
    />
  </a>
</li>
`;
